### PR TITLE
Support global aliases for commands

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Expand aliases when executing commands
- Add `@Botkube list aliases` command
- Modify `@Botkube list executors` to include aliases for executors
- Unify output of `@Botkube list sources` command
- Update help
    - Remove notice about required prefix
    - Add ability to list aliases
- Update E2E tests
- Add basic validation for aliases commands
    - I thought about checking executor names, but I wouldn't like to limit users. It's good to allow defining aliases for Botkube commands as well. Such validation would be too cumbersome for us to maintain (import cycles when it comes to cmd verbs) so I'd postpone this until someone reports an issue.

## TODO

- [ ] Update documentation - in a separate PR to `botkube-docs`.

### Help

image

### List executors
image

### List sources
image

#### Executing commands
image

image

### List aliases

image

image

image

### Filtering
image

## Testing

Check out the PR

Use the following config:
```yaml
plugins:
  repositories:
    botkube:
      url: https://storage.googleapis.com/botkube-plugins-latest/plugins-index.yaml

communications:
  "default-group":
    socketSlack:
      enabled: true
      appToken: "xapp-..."
      botToken: "xoxb-..."
      notification:
        type: "short"
      channels:
        "default":
          name: botkube-demo
          bindings:
            executors:
              - kubectl-read-only
              - helm
              - plugin-based
            sources:
              - k8s-all-events
              - k8s-recommendation-events
              - plugin-based

executors:
  'kubectl-read-only':
    kubectl:
      enabled: true
  helm:
     botkube/helm:
       config:
         helmCacheDir: /tmp/helm/.cache
         helmConfigDir: /tmp/helm/
         helmDriver: secret
       enabled: true
  'plugin-based':
    botkube/echo@v1.0.1-devel:
      enabled: false
      config:
        changeResponseToUpperCase: true

aliases:
  kc:
    command: kubectl
    displayName: "Kubectl alias"
  k:
    command: kubectl
    displayName: "Kubectl alias"
  kgp:
    command: kubectl get pods
    displayName: "Get pods"
  h:
    command: helm
    displayName: "Helm alias"
  e:
    command: echo
    displayName: "Echo alias"

sources:
  'plugin-based':
    displayName: "K8s ConfigMaps changes"
    botkube/cm-watcher:
      enabled: true
      config:
        configMap:
          name: cm-watcher-trigger
          namespace: botkube
          event: ADDED

settings:
  clusterName: "dev"
  upgradeNotifier: false
```

Run a few commands:

`@Botkube kgp -n botkube`
`@Botkube h version`
`@Botkube k get po`

Run `@Botkube list aliases`
Run `@Botkube list executors`

## Related issue(s)

https://github.com/kubeshop/botkube/issues/870

### CODE REVIEW

File: pkg/config/testdata/TestLoadedConfigValidationErrors/invalid-alias.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    command: helm version
+    displayName: "Helm version"
+  eee:
+    command: ""
+
+executors:
+  'kubectl-read-only':
+    kubectl:
+      enabled: true
+  helm:
```
Comment (josefkarasek) [inline pkg/config/testdata/TestLoadedConfigValidationErrors/invalid-alias.yaml:20]: Why are some keys wrapped in `'` and some aren't?

File: pkg/execute/alias.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	buf := new(bytes.Buffer)
+	w := tabwriter.NewWriter(buf, 5, 0, 1, ' ', 0)
+	fmt.Fprintln(w, "ALIAS\tCOMMAND\tDISPLAY NAME")
+	for _, aliasName := range maputil.SortKeys(aliasesToDisplay) {
+		aliasCfg := aliasesCfg[aliasName]
+		fmt.Fprintf(w, "%s\t%s\t%s\n", aliasName, aliasCfg.Command, aliasCfg.DisplayName)
+	}
+
+	if len(aliasesToDisplay) == 0 {
```
Comment (josefkarasek) [inline pkg/execute/alias.go:106]: Isn't this check redundant?

File: pkg/execute/executor.go [hunk 3]
```diff
@@ -78,11 +79,15 @@ func (flag CommandFlags) String() string {
...
+	rawCmd := sanitizeCommand(e.message)
+
+	expandedRawCmd := alias.ExpandPrefix(rawCmd, e.cfg.Aliases)
+	e.log.WithField("rawCmd", rawCmd).WithField("expandedRawCmd", expandedRawCmd).
+		Debugf("Expanding aliases from command...")
+
 	cmdCtx := CommandContext{
 		ClusterName:     e.cfg.Settings.ClusterName,
-		RawCmd:          rawCmd,
+		ExpandedRawCmd:  expandedRawCmd,
```
Comment (josefkarasek) [inline pkg/execute/executor.go:90]: :+1:

File: pkg/execute/kubectl_cmd_builder.go [hunk 4]
```diff
@@ -39,6 +38,8 @@ var knownCmdPrefix = map[string]struct{}{
 	filterPlaintextInputCommand:      {},
 }
 
+const kubectlName = "kubectl"
```
Comment (josefkarasek) [inline pkg/execute/kubectl_cmd_builder.go:41]: `"kubectl"` is defined already a few lines above

**Review by josefkarasek — APPROVED** [review]
Overall LGMT. With minor nits.

File: pkg/config/config.go [hunk 5]
```diff
@@ -209,7 +210,7 @@ type ActionBindings struct {
 type Sources struct {
 	DisplayName string           `yaml:"displayName"`
 	Kubernetes  KubernetesSource `yaml:"kubernetes"`
-	Plugins     PluginsExecutors `koanf:",remain"`
+	Plugins     PluginsMap       `koanf:",remain"`
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply pkg/config/config.go:213]: I know, but we have `Plugins` type already 😞 Any other ideas?

File: pkg/config/testdata/TestLoadedConfigValidationErrors/invalid-alias.yaml (hunk 1 above)
Comment (pkosiec) [reply pkg/config/testdata/TestLoadedConfigValidationErrors/invalid-alias.yaml:20]: It's mostly to distuingish configuration groups with other properties. I will unify them 👍

File: pkg/execute/alias.go (hunk 2 above)
Comment (pkosiec) [reply pkg/execute/alias.go:106]: You're right 👍 thanks

File: pkg/config/config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (josefkarasek) [reply pkg/config/config.go]: `Plugins       Plugins    `yaml:"plugins"`` -> `Plugins       PluginConfig    `yaml:"plugins"`` ?

**Pull request merged**